### PR TITLE
Add upgradeto action

### LINKED ISSUE #19: Upgradeto action to take the implementation address as parameter
Due to the subjective limit max-nonprivileged-inline-action-size of 4KB, it is difficult to make an inline action to deploy the implementation contract while not being rejected by the block producers. Because of this, we need to have an alternative way as follow:

1. deploy the implemenataion contract via normal pushtx action. Anyone can do this.
2. call upgradeto(impl_addr) instead of upgrade() 
3. call regtoken(...) to deploy the proxy contract with the impl_addr set by upgradeto

### PR DESCRIPTION
**Pull request by taokayan**
Resolve #19

### CODE REVIEW

**Review by yarkinwho — APPROVED** [review]
I am not sure about if we can do it or not but will std::swap work here for substitute in the address (that is, a vector)?

otherwise it looks fine. I will take care of the test after the IT is merged in.

File: antelope_contracts/contracts/erc20/src/erc20.cpp
Comment (arhag) [inline antelope_contracts/contracts/erc20/src/erc20.cpp:98]: It would be better to provide a more meaningful error message if someone calls `upgradeto` after the table entry with `id` of 0 was already created (e.g. due to a prior call of `upgradeto` or `upgrade`. Right now it goes deep into chainbase code before it would give an confusing error about a uniqueness constraint being violated.

I won't block this PR from being merged in because of this. But maybe @yarkinwho can add that check to this action with a better error message when he adds the tests in his PR (and includes in those tests checks that calling `upgrade` or `upgradeto` after they have already been called will fail as expected).

**Review by arhag — APPROVED** [review]

**Comment by arhag** [comment]
Also, please put a little bit more effort into the PR titles. They will be showing up in the release notes.

**Pull request merged**
